### DISCUSSION TITLE
Multistep plugin: Programatically trigger next page from another location

### DISCUSSION THREAD

**Discussion by MattFromGer** [formkit/formkit#685] [General]
I see that the multistep plugin has scoped slots, so you can pass a custom template for e.g. the next button. But what if the button is at another location and not the child component?

```
<FormKit type="multi-step">
  <FormKit type="step">
    <template #stepNext="{ handlers, node }">
      <!-- I don't want the button here -->
    </template>
  </FormKit>
</FormKit>

<nav>
  <FormKit type="button" @click="?">
    <!-- I want to call the action from here -->
  </FormKit>
</nav>
```

**Comment by fenilli** [comment]
You can get the multi-step node using `getNode` and there you have the handlers that you can call.

**Reply by MattFromGer** [reply]
A little bit hacky but it'll do, thank you!

**Comment by andrew-boyd** [comment]
With the upcoming `node.extend()` function in beta.17 we may be able to revisit this and have the node instance of type `multi-step` provide `node.nextStep()`, `node.prevStep()`, and `node.goToStep()` functions for better DX.

**Comment by MattFromGer** [comment]
@andrew-boyd Do you think you can incorporate this idea into the `multistep-updates` branch?

**Reply by andrew-boyd** [reply]
I think that's a swell idea

**Comment by andrew-boyd** [comment]
@MattFromGer Here's a playground of the current `next` tag showing external controls for the multistep. Something like this what you had in mind? https://formkit.link/0f59069efe97ce2fbde3920b9f5e6564

**Reply by MattFromGer** [reply]
Exactly. Maybe you can also expose properties like `isLast`, `isFirst`, in order to show/hide the corresponding buttons?
